static/src/javascripts/projects/common/modules/article/membership-events.js

```diff
@@ -0,0 +1,50 @@
+define([
+    'fastdom',
+    'qwery',
+    'Promise',
+    'common/utils/_',
+    'common/utils/$',
+    'common/utils/ajax-promise'
+], function (
+    fastdom,
+    qwery,
+    Promise,
+    _,
+    $,
+    ajax
+) {
+
+    var ELEMENT_INITIAL_CLASS = 'element-membership--not-upgraded',
+        ELEMENT_UPGRADED_CLASS = 'element-membership--upgraded';
+
+    function upgradeEvent(el) {
+        var href = $('a', el).attr('href'),
+            matches = href.match(/https:\/\/membership.theguardian.com/);
+
+        if (matches) {
+            return ajax({
```

Comment:
do you need to return anything at all? Then you wouldn't need to `Promise.resolve(null)` either